pyproject.toml

```diff
@@ -25,6 +25,12 @@ more-itertools = "^8.10.0"
 prometheus-client = ">=0.14.1,<0.15.0"
 pydantic = "^1.9.0"
 typing-extensions = "^4.2.0"
+boto3 = {version = "^1.24.2", optional = true}
+python-dotenv = {version = "^0.20.0", optional = true}
+
+[tool.poetry.extras]
+boto3 = ["boto3"]
```

Comment:
This, and the line below should be combined into

```python
s3 = ["boto3", "python-dotenv"]
```

Because they are both required to enable the S3 functionality

Reply:
Also boto3 is currently not optional - running with it not installed results in an error:

```
earhorn --help                                                                                    f35ef9e  
Traceback (most recent call last):    
  File "/home/kyle/.cache/pypoetry/virtualenvs/earhorn-9Oh9vI2L-py3.10/bin/earhorn", line 2, in <module>
    from earhorn.main import cli                                                                         
  File "/home/kyle/projects/earhorn/earhorn/main.py", line 13, in <module>
    from .stream_archive import (
  File "/home/kyle/projects/earhorn/earhorn/stream_archive.py", line 9, in <module>
    import boto3                      
ModuleNotFoundError: No module named 'boto3
```